lib/client.js

```diff
@@ -1494,14 +1498,34 @@ function write (client, request) {
     method === 'PATCH'
   )
 
+  if (util.isFormDataLike(body)) {
+    if (!extractBody) {
+      extractBody = require('./fetch/body.js').extractBody
+    }
+
+    const [bodyStream, contentType] = extractBody(body)
+    if (request.contentType == null) {
+      headers += `content-type: ${contentType}\r\n`
+    }
+    body = bodyStream.stream
+    contentLength = bodyStream.length
+  } else if (util.isBlobLike(body) && request.contentType == null && body.type) {
+    headers += `content-type: ${body.type}\r\n`
```

Comment:
Also... it's possible for body.type to be an empty string here. I don't think that's a valid header value?